api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java

```diff
@@ -147,6 +150,40 @@ public Order getOrder(Integer orderId) throws DAOException {
 		
 		return crit.list();
 	}
+
+	/**
+	 * @see org.openmrs.api.db.OrderDAO#getOrders(OrderSearchCriteria)
+	 */
+	@Override
+	public List<Order> getOrders(OrderSearchCriteria searchCriteria) {
+		Criteria crit = sessionFactory.getCurrentSession().createCriteria(Order.class);
+		
+		if (searchCriteria.getPatient() != null && searchCriteria.getPatient().getPatientId() != null) {
+			crit.add(Restrictions.eq("patient", searchCriteria.getPatient()));
+		}
+		if (searchCriteria.getCareSetting() != null && searchCriteria.getCareSetting().getId() != null) {
+			crit.add(Restrictions.eq("careSetting", searchCriteria.getCareSetting()));
+		}
+		if (searchCriteria.getConcepts() != null && !searchCriteria.getConcepts().isEmpty()) {
+			crit.add(Restrictions.in("concept", searchCriteria.getConcepts()));
+		}
+		if (searchCriteria.getOrderTypes() != null && !searchCriteria.getOrderTypes().isEmpty()) {
+			crit.add(Restrictions.in("orderType", searchCriteria.getOrderTypes()));
+		}
+		if (searchCriteria.getActivatedOnOrBeforeDate() != null) {
+			crit.add(Restrictions.le("dateActivated", searchCriteria.getActivatedOnOrBeforeDate()));
```

Comment:
I didn't think about this until now, but since we called it the "getActivateOnOrBeforeDate", we should be interpreting this as a Date (without a time component) even though a Java Data includes the time component.

For instance, if we have an Order that was activated on "2018-02-01 15:52:50", and in the search criteria you passed in "2018-02-01 00:00:00", I would expect that order to be returned, but I think as the query is written currently it wouldn't?

One solution would be in this query to change the time component of the getActivatedOnOrBeforeDate variable to be the last millisecond of the current date.  And then for the getActivatedOnOrAfterDate set the time component to 00:00:00

Reply:
Thanks @mogoodrich I just updated the PR